pages/developers/pep621.md

```diff
@@ -0,0 +1,148 @@
+    Programming Language :: Python :: 3.7
+    Programming Language :: Python :: 3.8
+    Programming Language :: Python :: 3.9
+    Programming Language :: Python :: 3.10
+    Topic :: Scientific/Engineering :: Physics
+
+[project.urls]
+Source = "https://github.com/scikit-hep/package"
+Documentation = "https://package.readthedocs.io/"
+Bug Tracker = "https://github.com/scikit-hep/package/issues"
+Discussions = "https://github.com/scikit-hep/package/discussions"
+Changelog = "https://package.readthedocs.io/en/latest/changelog.html"
+```
+
+You can read more about each field, and all allowed fields, in [PEP 621][],
+or [Flit](https://flit.readthedocs.io/en/latest/pyproject_toml.html#new-style-metadata), or
+[Whey](https://whey.readthedocs.io/en/latest/configuration.html).
+
+## Package structure
+
+All packages *should* have a `src` folder, with the package code residing
+inside it, such as `src/<package>/`.  This may seem like extra hassle; after
+all, you can type "`python`" in the main directory and avoid installing it if
+you don't have a `src` folder! However, this is a bad practice, and it causes
+several common bugs, such as running `pytest` and getting the local version
+instead of the installed version - this obviously tends to break if you build
+parts of the library or if you access package metadata.
+
+This sadly is not part of the standard metadata in `[project]`, so it depends
+on what backend you you use. Flit and PDM use automatic detection, while
+trampolim and whey do not, requiring a `tool` setting.
+
+
+## Versioning
+
+You can specify the version manually (as shown in the example), but the
+backends usually provide some automatic features to help you avoid this. Flit
+will pull this from a file if you ask it to. PDM can be instructed to look in a
+file or use git. Trampolim can be instructed to use git.
+
+
+## Including/excluding files in the SDist
+
+This it tool specific. [Flit info here](https://flit.readthedocs.io/en/latest/pyproject_toml.html#sdist-section).
+[PDM info here](https://pdm.fming.dev/pyproject/tool-pdm/#include-and-exclude-package-files). [Whey info here](https://whey.readthedocs.io/en/latest/configuration.html#tconf-tool.whey.additional-files).
+
+## Extras
+
+It is recommended to use extras instead of or in addition to making requirement
+files. These extras a) correctly interact with install requires and other
+built-in tools, b) are available directly when installing via PyPI, and c) are
+allowed in `requirements.txt`, `install_requires`, `pyproject.toml`, and most
+other places requirements are passed.
+
+Here is an example of a simple extras:
+
+```toml
+[project.optional-dependencies]
+test = [
+  "pytest >=6.0",
+]
+mpl = [
+  "matplotlib >=2.0",
+]
+```
+
+Self dependencies can be used by using the name of the package, such as `dev =
+["package[test,examples]"]`, but this requires Pip 21.2 or newer. We recommend
+providing at least `test`, `docs`, and `dev`.
```

Comment:
You mean test -> tests here, right?

Reply:
We generally use `[test]` for the extra, and `/tests` for the folder.

https://github.com/scikit-hep/boost-histogram/blob/928d066618ef45ddb3c6e64d515823a9a815520f/setup.py#L51-L64

Reply:
Indeed. Sorry, got the scope wrong. Agreed.